### PR TITLE
feat: reset device add config flow

### LINKED ISSUE #103: add multiple device should cache login info in memory
### Device type and model (or SN)

none

### The description of new feature

in this chanages:
https://github.com/wuwentao/midea_ac_lan/commit/1f56e69b7ebcabdf634a8eb14c44f44dec4470e7
we have removed write username/password to local json file, and not save username/password in local disk.

it caused a small issue: 
if user have more than 5-10+ device to be add, they should add each device and input login username/password every time.
we can cache it in memory and not write to file, can be done in future

### PR DESCRIPTION
**Pull request by wuwentao**
# PR Description
1.  cache login data
2. add remove cached login data step
3.  add a simple skip login option
4.  add async def check_cloud_login and check_cloud_key
5. reset add device step
6. disable v3 device add with v2/v1 protocol
7. force check device_id/device_protocol in finally step
8. all the error should be log in debug mode
9. to be done, code can be review via comments

## Related issue

fix #103 and more new feature support

### CODE REVIEW

**Review by chemelli74 — CHANGES_REQUESTED** [review]
Sorry to request changes, but I'm not a fan of a single PR with so many changes.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -63,3 +63,4 @@ repos:
         entry: pylint
         language: system
         types: [python]
+        args: ["--rcfile=pylintrc", "-v"]
```
Comment (chemelli74) [inline .pre-commit-config.yaml:66]: If this change is needed, please do it on a separate PR

**Review by chemelli74 — CHANGES_REQUESTED** [review]

**Comment by wuwentao** [comment]
> Sorry to request changes, but I'm not a fan of a single PR with so many changes.

thanks, it's a fullly of the old feature, not one feature, all the old process has been droped, and  most of the feature bounds in this new process.
and every step has add with all the detail comments and why we do it now, old process without any comments and info why it to do that...
if there is any error, just refer to the comments, please fullly drop old mode and old codes.

we don't have any testcase in integration side, most of the current process only can be test by manual action, and I have manual tested/verified with 90% of current code changes, confirmed it works as expected flow.

if we want to split it with more PRs, we should  consider how to deploy it step by step..... each feature should combind one by one, also conside how to replace the old code step by step.

wish you have any good solution to deloy it step by step with old proces replace, and I don't want to waste time with old process.... so I just select a fresh new one.
could you help me to finish it with more PRs?  I can drop my current PR, sorry....
waitting for your feedback with the solution when we got a agreement.
there is too many auth failed github issue, we should release a new version to check and confirm it.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (rokam) [inline .pre-commit-config.yaml:66]:
```suggestion
```

**Review by rokam — APPROVED** [review]

**Pull request merged**